fix: de/serialize tags as DisplayFromStr

aawsome opened the pull request:
Fixes a regression about de/serializing tags, introduced by #267

nardoor reviewed (commented):
1. I acknowledge this was a breaking change. Despite that, could there be some "test" that would have caught this "regression" ? 

2. A quick and naive look into #267 shows many similar `#[serd_as(...)]` suppression. I assume some might be intended. Could you make sure no similar regression were brought to other fields ?

aawsome commented:
About 1. The tests which did catch this regression (even though breaking, I changed them such that the desired config profiles should run) are located in rustic.

About 2. The tests did not find other things. The removal of `OneOrMany` was intentional, the problem for tags was the removal of `OneOrMany<DisplayFromStr>` which also removed the `DisplayFromStr` de/serialization...
